core/flow/tc_default.go

```diff
@@ -5,30 +5,47 @@ import (
 )
 
 type DirectTrafficShapingCalculator struct {
+	owner     *TrafficShapingController
 	threshold float64
 }
 
-func NewDirectTrafficShapingCalculator(threshold float64) *DirectTrafficShapingCalculator {
-	return &DirectTrafficShapingCalculator{threshold: threshold}
+func NewDirectTrafficShapingCalculator(owner *TrafficShapingController, threshold float64) *DirectTrafficShapingCalculator {
+	return &DirectTrafficShapingCalculator{
+		owner:     owner,
+		threshold: threshold,
+	}
 }
 
-func (d *DirectTrafficShapingCalculator) CalculateAllowedTokens(base.StatNode, uint32, int32) float64 {
+func (d *DirectTrafficShapingCalculator) CalculateAllowedTokens(uint32, int32) float64 {
 	return d.threshold
 }
 
-type DefaultTrafficShapingChecker struct {
-	rule *Rule
+func (d *DirectTrafficShapingCalculator) BoundOwner() *TrafficShapingController {
+	return d.owner
+}
+
+type RejectTrafficShapingChecker struct {
+	owner *TrafficShapingController
+	rule  *Rule
+}
+
+func NewRejectTrafficShapingChecker(owner *TrafficShapingController, rule *Rule) *RejectTrafficShapingChecker {
+	return &RejectTrafficShapingChecker{
+		owner: owner,
+		rule:  rule,
+	}
 }
 
-func NewDefaultTrafficShapingChecker(rule *Rule) *DefaultTrafficShapingChecker {
-	return &DefaultTrafficShapingChecker{rule: rule}
+func (d *RejectTrafficShapingChecker) BoundOwner() *TrafficShapingController {
+	return d.owner
 }
 
-func (d *DefaultTrafficShapingChecker) DoCheck(node base.StatNode, acquireCount uint32, threshold float64) *base.TokenResult {
-	if node == nil {
+func (d *RejectTrafficShapingChecker) DoCheck(resStat base.StatNode, acquireCount uint32, threshold float64) *base.TokenResult {
+	metricReadonlyStat := d.BoundOwner().boundStat.readOnlyMetric
+	if metricReadonlyStat == nil {
 		return nil
 	}
-	curCount := node.GetQPS(base.MetricEventPass)
+	curCount := metricReadonlyStat.GetQPS(base.MetricEventPass)
```

Comment:
Here we may need `GetSum` instead of `GetQPS` (i.e. avg per second).

Reply:
Yes, I will update here.